shared-bindings/time/__init__.c

```diff
@@ -248,7 +248,11 @@ STATIC mp_obj_t time_localtime(size_t n_args, const mp_obj_t *args) {
 
     mp_int_t secs = mp_obj_get_int(arg);
 
+    #if MICROPY_EPOCH_IS_1970
     if (secs < 0 || (mp_uint_t)secs < TIMEUTILS_SECONDS_1970_TO_2000) {
+    #else
+    if (secs < 0) {
+        #endif
```

Comment:
Should a conversion routine maybe be added to `lib/timeutils` that does this check, to isolate epoch checking in one place? I don't think lib/timeutils should necessarily throw an exception, but it could return success/failure on the conversion.